polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -388,15 +406,18 @@ void MLIRScanner::init(mlir::func::FuncOp function, const FunctionDecl *fd) {
   }
   Visit(stmt);
 
-  if (function.getFunctionType().getResults().size()) {
+  if (function.getResultTypes().size()) {
+    assert(!isa<gpu::GPUFuncOp>(function) &&
+           "SYCL kernel functions must always have a void return type.");
     mlir::Value vals[1] = {
         builder.create<mlir::memref::LoadOp>(loc, returnVal)};
     builder.create<ReturnOp>(loc, vals);
+  } else if (isa<gpu::GPUFuncOp>(function)) {
+    builder.create<gpu::ReturnOp>(loc);
   } else
```

Comment:
[nit: no need to enclose a single stmt in { }

Reply:
I think it just looks weird using braces in one branch and not in the following one (in fact, I'd rather add braces around the last branch).

Reply:
ok.